tests/integration-tests/tests/schedulers/test_slurm.py

```diff
@@ -1557,3 +1621,156 @@ def _test_memory_based_scheduling_enabled_true(
     # check RealMemory overridden via config file parameter
     assert_that(slurm_commands.get_node_attribute("queue1-dy-ondemand1-i3-1", "Memory")).is_equal_to("31400")
     # TODO: Add functional tests for memory-based scheduling
+
+
+def _test_scontrol_reboot_idle_nodes(
+    remote_command_executor,
+    slurm_commands,
+):
+    """Test scontrol reboot with idle nodes."""
+
+    # Get nodes and check that they are idle
+    nodes_in_queue = slurm_commands.get_compute_nodes("queue1")
+    assert_compute_node_states(slurm_commands, nodes_in_queue, "idle")
+
+    # Reboot nodes according to this logic:
+    # - 1 static idle node
+    # - 1 static idle node asap
+    # - 1 dynamic idle node
+    # - 1 dynamic idle node asap
+    static_nodes, dynamic_nodes = get_partition_nodes(nodes_in_queue)
+    assert_that(len(static_nodes)).is_equal_to(2)
+    assert_that(len(dynamic_nodes)).is_equal_to(2)
+    slurm_commands.reboot_compute_node(static_nodes[0], False)
+    slurm_commands.reboot_compute_node(static_nodes[1], True)
+    slurm_commands.reboot_compute_node(dynamic_nodes[0], False)
+    slurm_commands.reboot_compute_node(dynamic_nodes[1], True)
+
+    # Check that nodes enter a reboot state
+    time.sleep(2)
```

Comment:
1. Is this wait time fair enough or we risk to check the state too early?
2. Isn't the wait time to check transition already implicit in `wait_for_compute_nodes_states()`
1. I see you always wait 2 seconds before checking state transitions. What about having such wait time defined in a variable so that we have a single place to control the wait time?

Reply:
1. No, for such clusters 2 seconds should always be enough. The transition should be instantaneous, but I'm putting some buffer for safety.
2. We are doing an `assert_compute_node_states` before the `wait_for_compute_nodes_states`.
3. Addressing in the next commit.